src/components/common/DateRange.tsx

```diff
@@ -0,0 +1,82 @@
+import React, { useState } from 'react';
+import { DateRangePicker, defaultStaticRanges } from 'react-date-range';
+import 'react-date-range/dist/styles.css';
+import 'react-date-range/dist/theme/default.css';
+
+import { Box, Popover, TextField } from '@mui/material';
+
+import { format } from 'date-fns';
+import de from 'date-fns/locale/de';
+import enUS from 'date-fns/locale/en-US';
+import es from 'date-fns/locale/es';
+import fr from 'date-fns/locale/fr';
+
+import i18n, { useAnalyticsTranslation } from '@/config/i18n';
+
+// Mapping locales
+const locales: {
+  [key: string]: Locale;
+} = {
+  en: enUS,
+  fr: fr,
+  es: es,
+  de: de,
```

Comment:
we have arabic and italian as well!  Do you think this could go in the `i18next` file?